backend/src/baserow/core/two_factor_auth/handler.py

```diff
@@ -0,0 +1,127 @@
+from typing import cast
+
+from django.contrib.auth.models import AbstractUser
+from django.db.models import QuerySet
+
+from baserow.core.two_factor_auth.exceptions import (
+    TwoFactorAuthCannotBeConfigured,
+    TwoFactorAuthNotConfigured,
+    WrongPassword,
+)
+from baserow.core.two_factor_auth.registries import TwoFactorAuthProviderType
+
+from .models import TwoFactorAuthProviderModel
+from .registries import two_factor_auth_type_registry
+from .types import TwoFactorProviderForUpdate
+
+
+class TwoFactorAuthHandler:
+    def get_provider(
+        self, user: AbstractUser, base_queryset: QuerySet | None = None
+    ) -> TwoFactorAuthProviderModel | None:
+        """
+        Returns the user's provider from the database or None if no
+        provider is configured yet.
+
+        :param user: The user the provider is for.
+        :param base_queryset: The base queryset to use to build the query.
+        :return: The provider instance.
+        """
+
+        queryset = (
+            base_queryset if base_queryset else TwoFactorAuthProviderModel.objects.all()
+        )
+
+        provider = queryset.filter(user=user).first()
+        if provider is None:
+            return None
+
+        provider_specific: TwoFactorAuthProviderModel = provider.specific
+        return provider_specific
+
+    def get_provider_for_update(
+        self,
+        user: AbstractUser,
+    ) -> TwoFactorProviderForUpdate | None:
+        """
+        Returns the user's provider from the database or None if no
+        provider is configured yet.
+
+        :param user: The user the provider is for.
+        :return: The provider instance.
+        """
+
+        queryset = TwoFactorAuthProviderModel.objects.all().select_for_update(
+            of=("self",)
+        )
+        provider = self.get_provider(
+            user,
+            base_queryset=queryset,
+        )
+        if provider is None:
+            return None
+
+        return cast(
+            TwoFactorProviderForUpdate,
+            provider,
+        )
+
+    def configure_provider(
+        self,
+        provider_type_str: str,
+        user: AbstractUser,
+        **kwargs,
+    ) -> TwoFactorAuthProviderModel:
+        """
+        Configures the provider type for the user.
+
+        :param provider_type: The two-factor auth type of the provider.
+        :param user: The user configuring the authentication.
+        :param kwargs: Additional attributes of the provider.
+        :return: The created two-factor auth provider model.
+        """
+
+        if user.password == "":  # nosec
```

Comment:
What do you think of adding a small explanation on what it means when the password is empty, and why we don't want to allow settings a 2FA if the account only has SSO? For me, it makes perfect of sense, but someone newer looking at the code might not immediately know about this.